src/parser/tokenizer.h

```diff
@@ -212,6 +212,25 @@ struct Tokenizer {
     }
   }
 
+  Token ParseNumber(bool is_pos) {
+    std::stringstream ss;
+    while (More() && IsNumeric(Peek())) {
```

Comment:
out of the scope of this PR, but I imagine this is too loose in terms of accepting weird strings that satisfy the `IsNumeric` constraint, such as `1e2e3+..2`. That being said, `stod("1e2e3+..2")` returns `100`, so I guess it's alright lol.

Reply:
This is like some of the oldest tokenizer code, and I was just quickly trying to write one without getting super into regex, etc. Might be worth completely replacing soon.